packages/node/src/edge-functions/edge-handler.ts

```diff
@@ -196,20 +184,23 @@ export async function createEdgeEventHandler(
       redirect: 'manual',
       method: 'post',
       body: await serializeRequest(request),
+      //@ts-expect-error
+      headers: request.headers as HeadersInit,
```

Comment:
Are we sure we can pass Node's `IncomingMessage.Headers` type to Fetch's headers parameter without any kind of transform? I feel like the type casting shouldn't be necessary then

Reply:
Ideally, I want to remove that `@ts-expect-error` assertion. This how the error looks after removing it:

<img width="861" alt="CleanShot 2023-04-04 at 16 21 21@2x" src="https://user-images.githubusercontent.com/2096101/229822863-79fa8f9c-e1ca-4dcc-83ec-36942321f570.png">

Reply:
Hmm so its what I thought. TypeScript is being correct here. The fetch `headers` option is expecting type `HeadersInit`, and `request.headers` is only `Headers`. You can't just add `as HeadersInit` to get TypeScript to stop complaining. You would need to do `as unknown as HeadersInit` to properly type cast. 

But that doesn't answer my original question of, are you _certain_ that these can be casted like this? The answer can be yes, but you should add a detailed comment why (and then use the correct type cast i shared above).

Reply:
Fixed